app/routes/jokes/new.tsx

```diff
@@ -25,24 +19,17 @@ export const loader: LoaderFunction = async ({ request }) => {
   return json<UnpackData<typeof enforceUser>>(result.data);
 };
 
-type ActionData = ErrorResult & {
-  fields: {
-    name: string;
-    content: string;
-  };
-};
-const badRequest = (data: ActionData) => json(data, { status: 400 });
 export const action: ActionFunction = async ({ request }) => {
-  const fields = await inputFromForm(request);
-  const result = await createJoke(fields, await getUserId(request));
-  if (!result.success) {
-    return badRequest({ ...result, fields });
-  }
+  const result = await performMutation({
+    request,
+    schema: jokeSchema,
+    mutation: createJoke,
+  });
+  if (!result.success) return json(result, { status: 400 });
   return redirect(`/jokes/${result.data.id}?redirectTo=/jokes/new`);
```

Comment:
Check out [this commit](https://github.com/SeasonedSoftware/remix-forms/commit/309955a71c535c69aad3882533615e16d4d40eba). With `0.13.0` you can now use `formAction` instead of `performMutation` in this scenario, passing a function to `successPath`.